src/Corax/Queries/SortingMatches/SortingMultiMatch.Helpers.cs

```diff
@@ -0,0 +1,110 @@
+            (MatchCompareFieldType.Score, true) => SortBy<EntryComparerByScore>(orderMetadata),
+            (MatchCompareFieldType.Score, false) => SortBy<Descending<EntryComparerByScore>>(orderMetadata),
+            (MatchCompareFieldType.Spatial, true) => SortBy<EntryComparerBySpatial>(orderMetadata),
+            (MatchCompareFieldType.Spatial, false) => SortBy<Descending<EntryComparerBySpatial>>(orderMetadata),
+            (MatchCompareFieldType.Alphanumeric, true) => SortBy<EntryComparerByTermAlphaNumeric>(orderMetadata),
+            (MatchCompareFieldType.Alphanumeric, false) => SortBy<Descending<EntryComparerByTermAlphaNumeric>>(orderMetadata),
+            _ => &Fill<NullComparer, NullComparer, NullComparer>
+        };
+    }
+
+    private static delegate*<ref SortingMultiMatch<TInner>, Span<long>, int> SortBy<TComparer1>(OrderMetadata[] orderMetadata)
+        where TComparer1 : struct, IEntryComparer, IComparer<UnmanagedSpan>
+    {
+        if (orderMetadata.Length == 1)
+            return &Fill<TComparer1, NullComparer, NullComparer>;
+
+        return (orderMetadata[1].FieldType, orderMetadata[1].Ascending) switch
+        {
+            (MatchCompareFieldType.Integer, true) => SortBy<TComparer1, EntryComparerByLong>(orderMetadata),
+            (MatchCompareFieldType.Integer, false) => SortBy<TComparer1, Descending<EntryComparerByLong>>(orderMetadata),
+            (MatchCompareFieldType.Floating, true) => SortBy<TComparer1, EntryComparerByDouble>(orderMetadata),
+            (MatchCompareFieldType.Floating, false) => SortBy<TComparer1, Descending<EntryComparerByDouble>>(orderMetadata),
+            (MatchCompareFieldType.Sequence, true) => SortBy<TComparer1, EntryComparerByTerm>(orderMetadata),
+            (MatchCompareFieldType.Sequence, false) => SortBy<TComparer1, Descending<EntryComparerByTerm>>(orderMetadata),
+            (MatchCompareFieldType.Spatial, true) => SortBy<TComparer1,EntryComparerBySpatial>(orderMetadata),
+            (MatchCompareFieldType.Spatial, false) => SortBy<TComparer1, Descending<EntryComparerBySpatial>>(orderMetadata),
+            (MatchCompareFieldType.Alphanumeric, true) => SortBy<TComparer1, EntryComparerByTermAlphaNumeric>(orderMetadata),
+            (MatchCompareFieldType.Alphanumeric, false) => SortBy<TComparer1, Descending<EntryComparerByTermAlphaNumeric>>(orderMetadata),
+            _ => &Fill<TComparer1, NullComparer, NullComparer>
+        };
+    }
+
+    private static delegate*<ref SortingMultiMatch<TInner>, Span<long>, int> SortBy<TComparer1, TComparer2>(OrderMetadata[] orderMetadata)
+        where TComparer1 : struct, IEntryComparer, IComparer<UnmanagedSpan>
+        where TComparer2 : struct, IEntryComparer, IComparer<int>, IComparer<UnmanagedSpan>
+    {
+        if (orderMetadata.Length == 2)
+            return &Fill<TComparer1, TComparer2, NullComparer>;
+
+        return (orderMetadata[2].FieldType, orderMetadata[2].Ascending) switch
```

Comment:
How are you handling sorting by more? I'm not seeing the code for that?

Reply:
@maciejaszyk ping

Reply:
https://github.com/ravendb/ravendb/pull/16679/files#r1238519949

Reply:
https://github.com/ravendb/ravendb/pull/16679/files#r1243278957